### DISCUSSION TITLE
Label masks in LinkField class

### DISCUSSION THREAD

**Discussion by westguard** [UseDataKit/SDK#4] [Ideas]
I wonder if a mask for the href method would be helpful? For example:

`LinkField::create( 'submission_id', __('Edit Submission') )->sortable()->always_visible()->href_mask( 'https://mysite.com/wp-admin/submissions/view/?id=#value '),`

This would output:

`https://mysite.com/wp-admin/submissions/view/?id=123`

`#value` could perhaps be the ID of a column, e.g. `#submission_id`

**Comment by doekenorg** [comment]
@westguard yes, this is a great idea.👍

I actually forgot to port this when I moved over from JS to PHP (html) fields.  In the javascript helpers there is still a replace helper. The idea is to use a merge tag like structure. So `{field_name}`. In the actions this is also applied when calling a URL or AJAX call.

**Comment by westguard** [comment]
That sounds perfect!

**Comment by doekenorg** [comment]
@westguard Because every field has a `callback()` method to change the value on the field, you can create your desired result like this:

```php
$link = LinkField::create( 'id', 'Edit Submission' )
	->callback( fn( string $field, array $data ) => sprintf( 'https://wp-admin/submissions/view/?id=%d', $data[ $field ] ?? 0 ) )
	->with_label( 'Edit Submission #{id}' );
```
Would return 
```html
<a href="https://wp-admin/submissions/view/?id=123" target="_blank">Edit Submission #123</a>
```

Would that suffice?

**Comment by westguard** [comment]
Yes this is perfect! For WS Form I used:

```php
$link = LinkField::create( 'id', __('Edit Submission') )
	->callback( fn( string $field, array $data ) => sprintf('/wp-admin/admin.php?page=ws-form-submit&id=123#%d', $data[ $field ] ?? 0) )
	->with_label( __('Edit Submission #{id}') );
```

... where`123` is the form ID.
